### PR TITLE
task_3

### PR DESCRIPTION
**Pull request by AliaksandrRabtsau**
added serverless with ts
created 2 endpoints
https://j5ex5umy2j.execute-api.eu-west-1.amazonaws.com/dev/products

### CODE REVIEW

File: product-service/src/functions/getProductById/handler.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    const product = await getProductByIdTimeOut(productId);
+    return formatJSONResponse(product);
+  } catch (e) {
+    return formatJSONResponse({
+      error: e
+    });
+  }
+};
+
+const getProducts: () => Promise<Array<IProduct>> = async () => {
```
Comment (stasbereza) [inline product-service/src/functions/getProductById/handler.ts]: Here is no need to mark the function `getProducts` with the `async` keyword, as it has been already returning a Promise.

File: product-service/src/functions/getProductById/handler.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(products), 0);
+  })
+};
+
+const getProductByIdTimeOut: (productId: string | number) => Promise<IProduct> = async (productId) => {
```
Comment (stasbereza) [inline product-service/src/functions/getProductById/handler.ts:27]: As `async/await` is just a `syntax sugar` over the promises, there is no need to explicitly return a promise, that can be simplified like this. Does it work for you? 
```
const getProductByIdTimeOut: (productId: string | number) => IProduct = async (productId) => {
    const products = await getProducts();
    const product = products.find(p => p.id === productId);

    if (!product) {
       throw new Error(`Product "${productId}" not found`);
    }
    
   return products;
};
```

File: product-service/src/functions/getProducts/handler.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { IProduct } from 'src/models/Product';
+
+import schema from './schema';
+
+const getProducts: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async () => {
+  const products = await getProductsTimeOut();
+  return formatJSONResponse(products);
+};
+
+const getProductsTimeOut: () => Promise<Array<IProduct>> = async () => {
```
Comment (stasbereza) [inline product-service/src/functions/getProducts/handler.ts]: `async` can be removed here

**Comment by stasbereza** [comment]
There is a lot of stuff in your README, but I don't see the main: links to the Product Service API to test them. Sorry, maybe I missed something.

File: product-service/src/functions/getProductById/handler.ts (hunk 1 above)
Comment (AliaksandrRabtsau) [reply product-service/src/functions/getProductById/handler.ts]: removed

**Review by stasbereza — APPROVED** [review]

**Pull request merged**